cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/upgradesstable/UpgradeSSTableTask.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2016 Mesosphere
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mesosphere.dcos.cassandra.common.tasks.upgradesstable;
+
+
+import com.mesosphere.dcos.cassandra.common.tasks.CassandraDaemonTask;
+import com.mesosphere.dcos.cassandra.common.tasks.CassandraData;
+import com.mesosphere.dcos.cassandra.common.tasks.CassandraTask;
+import com.mesosphere.dcos.cassandra.common.tasks.CassandraTaskStatus;
+import org.apache.mesos.Protos;
+import org.apache.mesos.offer.TaskUtils;
+
+import java.util.Optional;
+
+/**
+ * UpgradeSSTableTask extends CassandraTask to implement upgrade SSTables. A
+ * CassandraDaemonTask must be running on the slave for a UpgradeSSTableTask to
+ * successfully execute.
+ * UpgradeSSTable rewrites older SSTables to the current version of Cassandra.
```

Comment:
We can re-write the comment part for old sstables to new, by your 2.X version to 3.X version cassandra. As this change in sstables is only done once.